src/QuartzNET-DynamoDB/JobStore.cs

```diff
@@ -325,32 +325,36 @@ public void StoreTrigger(IOperableTrigger newTrigger, bool replaceExisting)
 
         public bool RemoveTrigger(TriggerKey triggerKey)
         {
+            return RemoveTrigger(triggerKey, true);
+        }
+
+        private bool RemoveTrigger(TriggerKey triggerKey, bool removeOrphanedJob)
+        {
             bool found;
 
             lock (LockObject)
             {
-                var trigger = this.RetrieveTrigger(triggerKey);
+                var trigger = RetrieveTrigger(triggerKey);
                 found = trigger != null;
 
                 if (found)
                 {
                     _triggerRepository.Delete(triggerKey.ToDictionary());
 
-                    //todo: support orphaned job removal
-                    //					if (removeOrphanedJob)
-                    //					{
-                    //						IJobDetail jobDetail = this.RetrieveJob(trigger.JobKey);
-                    //						IList<IOperableTrigger> trigs = this.GetTriggersForJob(jobDetail.Key);
-                    //						if ((trigs == null
-                    //							|| trigs.Count == 0)
-                    //							&& !jobDetail.Durable)
-                    //						{
-                    //							if (this.RemoveJob(jobDetail.Key))
-                    //							{
-                    //								signaler.NotifySchedulerListenersJobDeleted(jobDetail.Key);
-                    //							}
-                    //						}
-                    //					}
+                    if (removeOrphanedJob)
+                    {
+                        IJobDetail jobDetail = RetrieveJob(trigger.JobKey);
+                        if (jobDetail != null) {
```

Comment:
You could have reversed the logic here and returned when null, to reduce nesting. Same below.